runtime/v2/logging/logging_windows.go

```diff
@@ -0,0 +1,99 @@
+// +build windows
+
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package logging
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"os"
+	"os/signal"
+	"syscall"
+
+	"github.com/Microsoft/go-winio"
+	"github.com/pkg/errors"
+)
+
+// Run the logging driver
+func Run(fn LoggerFunc) {
+	var err error
+	defer func() {
```

Comment:
In this case I think it would be better to create another func like `runInternal` (or an anonymous function) that actually returns an `error`, and then call that in `Run` and check its value explicitly to determine if we do the print and `os.Exit(1)`.

The issue with the current approach is if some code is added in the future that accidentally shadows `err` by redefining it in a nested scope and then returns, that `err` value won't be properly checked by the `defer`.

Reply:
yeah, agree 👍